scripts/benchmark_multithreaded.py

```diff
@@ -0,0 +1,218 @@
+#!/usr/bin/env python3
+
+# To prevent _tkinter.TclError: https://stackoverflow.com/a/37605654
+import matplotlib
+matplotlib.use('Agg')
+import matplotlib.pyplot as plt
+
+import os
+import argparse
+import multiprocessing
+import subprocess
+import json
+
+from datetime import datetime
+
+
+# This script executes a Hyrise benchmark multiple times over a specified range of cores, and produces a plot
+# of the results. You'll have to provide the path to the benchmark executable (e.g., hyriseBenchmarkTPCH)
+# as argument, as well as all parameters you want to be passed to the executable, like --scale, --chunk_size, etc.
+# You'll also have to specify the number of query runs as either a fixed number for all cores (--fixed-runs),
+# or a number of runs per core (--runs-per-core).
+# You can also specify the cores to be benchmarked, otherwise a default range will be used.
+#
+# Example usage 1:
+# python3 ./scripts/benchmark_multithreaded.py -e ./build-release/hyriseBenchmarkTPCH --runs 1000 -v
+#
+# Example usage 2:
+# python3 ./scripts/benchmark_multithreaded.py -e ./build-release/hyriseBenchmarkTPCH --runs 1000 -v --scale 1 --cores 10 20 30 --queries 1,3,6,12 --clients 20
+
+
+MAX_CORE_COUNT = max(1, multiprocessing.cpu_count() - 1)  # Keep one core empty, same as in topology.cpp
```

Comment:
Thanks for your changes! I agree with everything except this one. The Topology already handles this if we are on a non-NUMA system. On a NUMA-system it uses all available CPUs. I don't think we should restrict this in two different spots, if it has to go through one of them anyway. Also this would easily get overridden when specifying --cores or --max-cores, while still possibly being changed by the Topology.

Reply:
Okay, that makes sense. After thinking about it, I honestly think we should remove this limitation completely. If I want to benchmark something on my laptop, with 4 cores, and go `hyriseBenchmarkTPCH --cores 4 ...`, then I want to benchmark it with 4 cores, and not with 3. If I want to leave one core out because I want to work on something else while benchmarking, then I can specify that manually. We don't restrict CPUs on a NUMA system either, if we say use 80 cores, we do it.
Basically I don't want the Benchmark to "not allow" me to fully utilize my system for benchmarking if I want to... right?

Reply:
Which btw, funnily, I did the whole time before. Since I installed libnuma some time ago to test something, CMake thinks I'm on a NUMA system, so I never even got that limitation when benchmarking on my laptop, because I always had a "NUMA" Topology with only 1 node and 4 cores. Worked perfectly, no freeze. Of course I didn't do anything else during that benchmark, but that was my intent, because I knew that all 4 cores would be used.
That's why I forgot about that limitation, because it didn't come up when I tested it ^^

Reply:
Agreed. Please get rid of it.